Automattic/wp-calypso
# Allow removal of disconnected Jetpack purchases

Due the UI preventing any management of purchases if the user has no sites, an edge case
happens where a user who has no sites, made a Jetpack purchase and then disconnected
that Jetpack purchase is unable to manage their purchase. This will move the check that
sites exist to after the call to fetch user purchases, allowing all purchases to be shown,
regardless of the number of sites to which a user belongs. For listing purchases, if there
are no purchases found, it will show the no sites message. Managing purchases have the
check for user sites removed entirely as manage purchases will redirect to /me/purchases if
something is wrong.

#### Changes proposed in this Pull Request

* For `/me/purchases` move the `userHasNoSites` check from the controller to inside the `PurchasesList` component after no purchases have been found.
* For `/me/purchases/:site/:purchaseId` remove the check for `userHasNoSites` entirely. The component will redirect to `/me/purchases` if an invalid combination of `:site` and `:purchaseId` are encountered.

#### Testing instructions



* On a brand new user with no existing sites, purchase a Jetpack subscription
* Disconnect the purchased Jetpack subscription
* View this branch and navigate to `/me/purchase` and if everything works properly you should be able to remove your Jetpack subscription (if you are outside of the refund window) or cancel & refund your Jetpack subscription (if you are within the refund window)
